### ISSUE TITLE
BOM completeness checks

### ISSUE THREAD

**Issue by ppalaga** [quarkusio/quarkus#26067]
### Describe the bug

https://github.com/quarkusio/quarkus/pull/26026 seems to suggest that the completeness of the BOM is not checked on a regular basis. 

This could perhaps be done during the BOM flattening.

I just found out that `quarkus-smallrye-reactive-messaging-kotlin-deployment` is also missing. 
It is hard to figure out manually whether anything else is missing.

cc @aloubyansky

**Comment by aloubyansky** [comment]
@ppalaga by completeness you mean for every runtime artifact there a deployment artifact is expected to be present in the BOM?

**Comment by ppalaga** [comment]
> by completeness you mean for every runtime artifact there a deployment artifact is expected to be present in the BOM?

Yes, at least. 

Maybe other related checks are possible:

* All modules (and their io.quarkus transitives) under `extensions` directory are managed
* All BOM entries exist in some of the configured remote repos

**Comment by aloubyansky** [comment]
> * All BOM entries exist in some of the configured remote repos

AFAICT, there will pretty much always be some non-existing constraints imported with thirdparty BOMs.

**Comment by ppalaga** [comment]
> AFAICT, there will pretty much always be some non-existing constraints imported with thirdparty BOMs.

Yep, but isn't flattening a good occasion to remove those?

**Comment by aloubyansky** [comment]
It is. Although it'll increase the build time. We'd have to disable it for most common build-for-dev use-cases. Otherwise our colleagues will hate it.

**Comment by ppalaga** [comment]
> It is. Although it'll increase the build time. We'd have to disable it for most common build-for-dev use-cases. Otherwise our colleagues will hate it.

Yes, it does not have to happen on every build. I am just asking for doing it regularly.

**Comment by gsmet** [comment]
Could we just have a check in the enforcer that makes sure all `io.quarkus` dependencies don't have a version set? That would then require them to be defined in the BOM (or in a dependency management block in the parent for the independent projects).
My guess is that it would be easy to implement given we already have some additional enforcer rules and fast.
